numba/datamodel/models.py

```diff
@@ -491,6 +490,17 @@ def from_data(self, builder, value):
         return NotImplemented
 
 
+@register_default(types.NestedArray)
+class NestedArrayModel(ArrayModel):
+    def __init__(self, dmm, fe_type):
+        self._be_type = dmm.lookup(fe_type.dtype).get_data_type()
+        super(NestedArrayModel, self).__init__(dmm, fe_type)
+
+    def get_data_type(self):
```

Comment:
Intuitively, the data type of an array is an array of the element's data type, so I don't understand this one. Where is the data type necessary in this PR?